src/common/k8s-api/api-manager/api-manager.ts

```diff
@@ -33,20 +33,38 @@ export class ApiManager {
   private readonly externalApis = observable.array<KubeApi>();
   private readonly externalStores = observable.array<KubeObjectStore>();
 
-  constructor(private readonly dependencies: Dependencies) {}
+  private readonly apis = observable.map<string, KubeApi>();
+
+  constructor(private readonly dependencies: Dependencies) {
+    // NOTE: this is done to preserve the old behaviour of an API being discoverable using all previous apiBases
+    autorun(() => {
+      const apis = chain(this.dependencies.apis.get().values())
+        .concat(this.externalApis.values());
+      const removedApis = new Set(this.apis.values());
+
+      for (const api of apis) {
+        removedApis.delete(api);
+        this.apis.set(api.apiBase, api);
+      }
+
+      for (const api of removedApis) {
+        for (const [apiBase, storedApi] of this.apis) {
+          if (storedApi === api) {
+            this.apis.delete(apiBase);
+          }
+        }
+      }
+    });
+  }
```

Comment:
Not unit testing code like this strikes terror in my heart.

Reply:
This is though, all of the tests relating to drawers go through this code.